src/auth/OAuthAuthenticator.js

```diff
@@ -206,57 +206,62 @@ OAuthAuthenticator.prototype.passwordGrant = function(userData, options, cb) {
 };
 
 /**
- * Sign in using a refresh token
+ * Exchange a refresh token
  *
  * @method    refreshToken
  * @memberOf  module:auth.OAuthAuthenticator.prototype
  *
  * @example <caption>
  *   Given a refresh token from a previous authentication request
- *   it will return a JSON with the access_token and id_token.
+ *   it will return a JSON with the access_token and id_token if
+ *   the openid scope was originally included.
  *   More information in the
  *   <a href="https://auth0.com/docs/api/authentication#refresh-token">
  *     API Docs
  *   </a>.
  * </caption>
  *
  * var data = {
- *   client_id: '{CLIENT_ID}', // Optional field.
  *   refresh_token: '{REFRESH_TOKEN}',
  * };
  *
- * auth0.oauth.refreshToken(data, function (err, userData) {
+ * auth0.oauth.refreshToken(data, function (err, data) {
  *   if (err) {
  *     // Handle error.
  *   }
  *
- *   console.log(userData);
+ *   console.log(data);
  * });
  *
- * @param   {Object}    userData                User credentials object.
- * @param   {String}    userData.refresh_token  Refresh token.
+ * @param   {Object}    data                Data object.
+ * @param   {String}    data.refresh_token  Refresh token.
  *
  * @return  {Promise|undefined}
  */
-OAuthAuthenticator.prototype.refreshToken = function(userData, cb) {
-  var params = {
-    type: 'token'
-  };
+OAuthAuthenticator.prototype.refreshToken = function(data, cb) {
+  if (!data || typeof data !== 'object') {
+    throw new ArgumentError('Missing data object');
+  }
+
   var defaultFields = {
     client_id: this.clientId,
+    client_secret: this.clientSecret,
```

Comment:
Node should be running on a server so we should always be sending credentials to the token endpoint

Reply:
Should we clarify that both in the doc blocks for this method and in the readme?

e.g. for readme we can say 
> Node.js client library for the Auth0 platform.

e.g. Node.js **server side** client library for the Auth0 platform.

Reply:
Actually, I think I spoke too fast ... this could be used in a CLI app, in which case we wouldn't have a secret. This will still work, though, as the token endpoint should be configured correctly.